File: java/unidad_03/Problema_3_11/src/Problema_3_11.java
import java.util.InputMismatchException;
import java.util.Scanner;

/*
 * El programa, obtiene el total de votos y el porcentaje obtenido por los cuatro
 * candidatos de una elección.
 */

public class Problema_3_11 {

    public static void main(String[] args) {

        int voto;

        int votosCandidato01 = 0;
        int votosCandidato02 = 0;
        int votosCandidato03 = 0;
        int votosCandidato04 = 0;

        float totalVotos;

        float porcentajeCandidato01;
        float porcentajeCandidato02;
        float porcentajeCandidato03;
        float porcentajeCandidato04;

        try (Scanner scanner = new Scanner(System.in)) {

            System.out.print("Voto: ");
            voto = scanner.nextInt();

            while (voto != 0) {

                switch (voto) {

                    case 1:

                        votosCandidato01 += 1;
                        break;

                    case 2:

                        votosCandidato02 += 1;
                        break;

                    case 3:

                        votosCandidato03 += 1;
                        break;

                    case 4:

                        votosCandidato04 += 1;
                        break;

                    default:

                        System.out.println("Error: voto no válido.");
                        break;
                }
                System.out.print("Voto: ");
                voto = scanner.nextInt();
            }
        } catch (InputMismatchException e) {

            System.out.println(e);
            return;
        }
        totalVotos = votosCandidato01 + votosCandidato02 + votosCandidato03 + votosCandidato04;

        porcentajeCandidato01 = (votosCandidato01 / totalVotos) * 100;
        porcentajeCandidato02 = (votosCandidato02 / totalVotos) * 100;
        porcentajeCandidato03 = (votosCandidato03 / totalVotos) * 100;
        porcentajeCandidato04 = (votosCandidato04 / totalVotos) * 100;

        System.out.printf("Votos candidato 01: %d %nPorcentaje: %.2f %n", votosCandidato01, porcentajeCandidato01);
        System.out.printf("Votos candidato 02: %d %nPorcentaje: %.2f %n", votosCandidato02, porcentajeCandidato02);
        System.out.printf("Votos candidato 03: %d %nPorcentaje: %.2f %n", votosCandidato03, porcentajeCandidato03);
        System.out.printf("Votos candidato 04: %d %nPorcentaje: %.2f %n", votosCandidato04, porcentajeCandidato04);

        System.out.printf("Total de votos: %.0f", totalVotos);
    }
}

File: python/unidad_03/problema_3_11.py
'''
El programa, obtiene el total de votos y el porcentaje obtenido por los cuatro
candidatos de una elección.
'''

def main():

    votos_candidato_01 = 0
    votos_candidato_02 = 0
    votos_candidato_03 = 0
    votos_candidato_04 = 0

    try:

        voto = int(input("Voto: "))

        while (voto != 0):

            if voto == 1:

                votos_candidato_01 += 1

            elif voto == 2:

                votos_candidato_02 += 1

            elif voto == 3:

                votos_candidato_03 += 1

            elif voto == 4:

                votos_candidato_04 += 1

            else:

                print("Error: voto no válido.")

            voto = int(input("Voto: "))
        
    except ValueError as e:

        print(e)
        return

    total_votos = votos_candidato_01 + votos_candidato_02 + votos_candidato_03 + votos_candidato_04

    porcentaje_candidato_01 = (votos_candidato_01 / total_votos) * 100
    porcentaje_candidato_02 = (votos_candidato_02 / total_votos) * 100
    porcentaje_candidato_03 = (votos_candidato_03 / total_votos) * 100
    porcentaje_candidato_04 = (votos_candidato_04 / total_votos) * 100

    print(f"Votos candidato 01: {votos_candidato_01} \nPorcentaje: {porcentaje_candidato_01:.2f}")
    print(f"Votos candidato 02: {votos_candidato_02} \nPorcentaje: {porcentaje_candidato_02:.2f}")
    print(f"Votos candidato 03: {votos_candidato_03} \nPorcentaje: {porcentaje_candidato_03:.2f}")
    print(f"Votos candidato 04: {votos_candidato_04} \nPorcentaje: {porcentaje_candidato_04:.2f}")

    print(f"Total de votos: {total_votos}")

if __name__ == '__main__':

    main()